src/main/java/Interactors/IOController.java

```diff
@@ -68,18 +68,166 @@ public void connectLogic(){
      * Helper function that takes state data and formulates the data into a string readable by the user
      */
     public void displayOptions(){
+    //    StringBuilder outputString = new StringBuilder();
+ //       outputString.append(currentState.getDescription());
+        this.output = stateToString(currentState);
+
+    }
+    public String stateToString(State state){
         StringBuilder outputString = new StringBuilder();
```

Comment:
Perhaps have some comments on each one, but not it should be good in that it adheres to clean architecture. Also, we will need to shorten this using helper classes and also some other refactoring.